implementations/rust/ockam/ockam_vault/src/traits/secrets_store.rs

```diff
@@ -0,0 +1,178 @@
+use crate::{PublicKey, Secret, SecretAttributes, StoredSecret};
+use ockam_core::{async_trait, compat::boxed::Box, KeyId, Result};
+
+/// This traits provides all the functionalities related to the management of secrets
+#[async_trait]
+pub trait SecretsStore:
+    EphemeralSecretsStore + PersistentSecretsStore + SecretsStoreReader + Sync + Send
+{
+}
+
+impl<T: EphemeralSecretsStore + PersistentSecretsStore + SecretsStoreReader + Sync + Send>
+    SecretsStore for T
+{
+}
+
+/// This traits supports the creation / retrieval / deletion of ephemeral secrets
+#[async_trait]
+pub trait EphemeralSecretsStore: SecretsStoreReader + Sync + Send {
+    /// Generate a secret and persist it to ephemeral memory
+    async fn create_ephemeral_secret(&self, attributes: SecretAttributes) -> Result<KeyId>;
+    /// Import a secret and persist it to ephemeral memory
+    async fn import_ephemeral_secret(
+        &self,
+        secret: Secret,
+        attributes: SecretAttributes,
+    ) -> Result<KeyId>;
+    /// Export an ephemeral secret
+    /// Use the description in an error message if the secret cannot be found
+    async fn get_ephemeral_secret(&self, key_id: &KeyId, description: &str)
```

Comment:
Adding description here just for better error message looks weird to me

Reply:
I did this because I found it quite helpful to get a specific error message rather than "key not found". But if you feel it's too much, I can revert this.